src/psyclone/psyir/nodes/reference.py

```diff
@@ -102,10 +102,17 @@ def is_read(self):
         '''
         # pylint: disable=import-outside-toplevel
         from psyclone.psyir.nodes.assignment import Assignment
+        from psyclone.psyir.nodes.intrinsic_call import IntrinsicCall
         parent = self.parent
         if isinstance(parent, Assignment):
             if parent.lhs is self:
                 return False
+
+        # If we have an intrinsic call parent then we need to check if its
+        # an inquiry. Inquiry functions don't read from their arguments.
+        if isinstance(parent, IntrinsicCall) and parent.is_inquiry:
+            return False
```

Comment:
They read from some arguments, e.g. var1 will be read. If that's not a problem for now just put a TODO to 2302 as we need more information to do it better.
`UBOUND(ARRAY, DIM=var1)`

Reply:
@sergisiso is it always the case that argument 1 is the inquiry and the other arguments are read-only?

Reply:
Ah I just read 2302 - yeah its not causing issues for now - if we have no idea on which argument is the inquiry then its difficult and I have concerns about this.